validator/rpc/wallet.go

```diff
@@ -231,6 +233,45 @@ func (s *Server) ImportKeystores(
 	}, nil
 }
 
+// ValidateKeystores checks whether a set of EIP-2335 keystores in the request
+// can indeed be decrypted using a password in the request. If there is no issue,
+// we return an empty response with no error. If the password is incorrect for a single keystore,
+// we return an appropriate error.
+func (s *Server) ValidateKeystores(
+	ctx context.Context, req *pb.ValidateKeystoresRequest,
+) (*ptypes.Empty, error) {
+	if req.KeystoresPassword == "" {
+		return nil, status.Error(codes.InvalidArgument, "Password required for keystores")
+	}
+	// Needs to unmarshal the keystores from the requests.
+	if req.Keystores == nil || len(req.Keystores) < 1 {
+		return nil, status.Error(codes.InvalidArgument, "No keystores included in request")
+	}
+	decryptor := keystorev4.New()
+	for i := 0; i < len(req.Keystores); i++ {
+		encoded := req.Keystores[i]
+		keystore := &keymanager.Keystore{}
+		if err := json.Unmarshal([]byte(encoded), &keystore); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "Not a valid EIP-2335 keystore JSON file: %v", err)
+		}
+		if _, err := decryptor.Decrypt(keystore.Crypto, req.KeystoresPassword); err != nil {
+			doesNotDecrypt := strings.Contains(err.Error(), "invalid checksum")
```

Comment:
This is fragile. Can you use some constant error to compare? 
If the returned error were ever changed then this logic would break.